File: laboratory5c++/3.cpp
#include <iostream>
#include <random>
using namespace std;

int positive_edit(int number, int modified)
{
    return number + modified;
}

int negative_edit(int number, int modified)
{
    return number - modified;
}

int* random_massive(int start, int end, int len) {
    int* numbers = new int[len];

    random_device rd;
    mt19937 gen(rd());
    uniform_int_distribution<> distrib(start, end);

    for (int i = 0; i < len; i++) {
        numbers[i] = distrib(gen);
    }

    return numbers; 
}

void print_massive(int* massive, int size)
{
    for(int i = 0; i < size; i++)
    {
        cout << *(massive+i) << " ";
    }
    cout << endl;
}

void modifyArray(int* massive, int size)
{
    for(int i = 0; i < size; i++)
    {
        if (*(massive+i) > 0){
            *(massive+i) += 10;
        }
        else if (*(massive+i) < 0){
            *(massive+i) -= 5;
        }
    }
}

int main()
{
    int size = 10;
    int* numbers = random_massive(-500,500,size);
    print_massive(numbers, size);
    
    modifyArray(numbers, size);
    
    print_massive(numbers, size);    
    return 0;
}

File: pr/laboratory2go/3.go
package main

import "fmt"

func sum(numbers []int, number int) int {
	sum := 0
	for i := 0; i < len(numbers); i++ {
		if number < numbers[i] {
			sum += numbers[i]
		}
	}
	return sum
}

func main() {
	numbers := [5]int{2, 2, 1, 4, 10}
	number := 4
	fmt.Println("Сумма элементов, больше определенного элемента:", sum(numbers[:], number))
}
